src/mscorlib/corefx/System/Globalization/CultureInfo.Unix.cs

```diff
@@ -13,14 +13,20 @@ public partial class CultureInfo : IFormatProvider
         /// </remarks>
         private static CultureInfo GetUserDefaultCultureCacheOverride()
         {
-            // TODO: Implement this fully.
             return null;
         }
 
-        private static CultureInfo GetUserDefaultCulture()
+        internal static CultureInfo GetUserDefaultCulture()
         {
-            // TODO: Implement this fully.
-            return CultureInfo.InvariantCulture;
+            CultureInfo temp = null;
```

Comment:
Changed to use Invariant in case of fail on GetDefaultLocaleName. I don't think that value should ever be null. The CurrentCulture property is settable and doesn't allow null, although there is one check for null in a specific area but that seems to be defensive